src/sfizz/Wavetables.h

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: BSD-2-Clause
+
+// This code is part of the sfizz library and is licensed under a BSD 2-clause
+// license. You should have receive a LICENSE.md file along with the code.
+// If not, contact the sfizz maintainers at https://github.com/sfztools/sfizz
+
+#pragma once
+#include <absl/types/span.h>
+#include <memory>
+#include <complex>
+
+namespace sfz {
+
+class WavetableMulti;
+
+/**
+   An oscillator based on wavetables
+ */
+class WavetableOscillator {
+public:
+    /**
+       Initialize with the given sample rate.
+       Run it once after instantiating.
+     */
+    void init(double sampleRate);
+
+    /**
+       Reset the oscillation to the initial phase.
+     */
+    void clear();
+
+    /**
+       Set the wavetable to generate with this oscillator.
+     */
+    void setWavetable(const WavetableMulti* wave);
+
+    /**
+       Compute a cycle of the oscillator, with constant frequency.
+     */
+    void process(float frequency, float* output, unsigned nframes);
+
+    /**
+       Compute a cycle of the oscillator, with varying frequency.
+     */
+    void processModulated(const float* frequencies, float* output, unsigned nframes);
+
+private:
+    /**
+       Interpolate a value from a part of table, with delta in 0 to 1 excluded.
+       There are `TableExtra` elements available for reading.
+       (cf. WavetableMulti)
+     */
+    static float interpolate(const float* x, float delta);
```

Comment:
This a function where you can introduce an interpolation function of choice if you'd like to test others.

Since it has `tableExtra=4`, you could do interpolations to the 4th order.
In fact `tableExtra` can be set to 1 if it is kept linear only.

(linear can be considered sufficient for wavetables, they are all lowpassed in advance)

Reply:
Ok since it kinda depends on the table structure let's keep it here for now.